core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -1138,11 +1138,9 @@ public void readFields(DataInput in) throws IOException {
   }
 
   private void deserializeIndexMetadata() throws IOException {
-    if (indexMetadata == null) {
-      String indexMeta = tableInfo.getFactTable().getTableProperties().get(getTableId());
-      if (null != indexMeta) {
-        indexMetadata = IndexMetadata.deserialize(indexMeta);
-      }
+    String indexMeta = tableInfo.getFactTable().getTableProperties().get(getTableId());
```

Comment:
deserializing metadata everytime will impact the performance, so please reload the index table cache  during Alter rename operation itself

Reply:
Done